sway/desktop/output.c

```diff
@@ -217,9 +218,26 @@ static void render_texture(struct wlr_output *wlr_output,
 	pixman_region32_fini(&damage);
 }
 
+static bool surface_is_popup(struct wlr_surface *surface) {
+	if (wlr_surface_is_xdg_surface(surface)) {
+		struct wlr_xdg_surface *xdg_surface =
+			wlr_xdg_surface_from_wlr_surface(surface);
+		return xdg_surface->role == WLR_XDG_SURFACE_ROLE_POPUP;
+	}
+	if (wlr_surface_is_xdg_surface_v6(surface)) {
+		struct wlr_xdg_surface_v6 *xdg_surface_v6 =
+			wlr_xdg_surface_v6_from_wlr_surface(surface);
+		return xdg_surface_v6->role == WLR_XDG_SURFACE_V6_ROLE_POPUP;
+	}
+	return false;
+}
+
 static void render_surface_iterator(struct wlr_surface *surface, int sx, int sy,
 		void *_data) {
 	struct render_data *data = _data;
+	if (data->popups != surface_is_popup(surface)) {
```

Comment:
This will render subsurfaces of popups when `data->popups == false`.

Reply:
Do you know of any program that does this which I can use for testing?

Reply:
No, sorry